dali/pipeline/data/tensor_list.h

```diff
@@ -626,7 +596,128 @@ class DLL_PUBLIC TensorList : private Buffer<Backend> {
     meta_[idx] = meta;
   }
 
+  // Reexpose all public Buffer functions apart from contiguous buffer accessors.
+  // TensorList is being reworked to sample-only access and this is intermediate step
+  // that prevents reintroducing that access in any of DALI operators
+
+
+  /**
+   * @brief Returns the TypeInfo object that keeps track of the
+   * datatype of the underlying storage.
+   */
+  const TypeInfo &type_info() const {
+    return data_.type_info();
+  }
+
+  /**
+   * @brief Returns the id of the datatype of the underlying storage.
+   */
+  DALIDataType type() const {
+    return data_.type();
+  }
+
+  /**
+   * @brief Returns the size in bytes of the underlying data
+   */
+  size_t nbytes() const {
+    return data_.nbytes();
+  }
+
+  /**
+   * @brief Returns the real size of the allocation
+   */
+  size_t capacity() const {
+    return data_.capacity();
+  }
+
+  /**
+   * @brief Set the type of the TensorList. The type needs to be set before calling
+   * the Resize function that gives the shape. Type can be changed, if the current storage
+   * is not big enough, the memory will be reallocated.
+   */
+  inline void set_type(const DALIDataType new_type_id) {
+    data_.set_type(new_type_id);
+  }
+
+  /**
+   * @brief Set the type of the TensorList. The type needs to be set before calling
+   * the Resize function that gives the shape. Type can be changed, if the current storage
+   * is not big enough, the memory will be reallocated.
+   */
+  template <typename T>
+  inline void set_type() {
+    data_.set_type(TypeTable::GetTypeId<T>());
+  }
+
+  /**
+   * @brief Sets the type of allocation (pinned/non-pinned) for CPU TensorList
+   */
+  inline void set_pinned(bool pinned) {
+    data_.set_pinned(pinned);
+  }
+
+  /**
+   * @brief Returns the type of allocation (pinned/non-pinned) for CPU TensorList
+   */
+  bool is_pinned() const {
+    return data_.is_pinned();
+  }
+
+   /**
+   * @brief Returns a device this TensorList was allocated on
+   * If the backend is CPUBackend, return -1
+   */
+  int device_id() const {
+    return data_.device_id();
+  }
+
+  /**
+   * @brief Sets a device this TensorList was allocated on
+   * If the backend is CPUBackend, should be -1
+   */
+  void set_device_id(int device) {
+    data_.set_device_id(device);
+  }
+
+  /**
+   * @brief Return true if there was data allocation
+   */
+  inline bool has_data() const noexcept {
+    return data_.has_data();
+  }
+
+  /**
+   * @brief Returns a bool indicating if the list shares its underlying storage.
+   */
+  inline bool shares_data() const {
+    return data_.shares_data();
+  }
+
+  /**
+   * @brief Sets a custom allocation function.
+   *
+   * Sets a custom allocation function. The allocation function returns
+   * a shared pointer with a matching deleter.
+   *
+   * @remarks Experimental - subject to change
+   */
+  inline void set_alloc_func(typename Buffer<Backend>::AllocFunc allocate) {
```

Comment:
Is it really experimental? Or just temporary?

Reply:
I just propagated the docstring from what we have in Buffer.